compiler/src/main/resources/templates/ecr_docker_preamble.ssp

```diff
@@ -27,14 +27,15 @@
     # disable echo to avoid logging password
     set +x
 
-    # get the password from the aws client
-    password=${bashDollar}(awscliv2 ecr get-login-password --region ${bashDollar}{AWS_REGION})
-
     # Log into docker. Retry several times.
     i=0
     while [[ ${bashDollar}i -le 5 ]]; do
         # don't exit if docker command fails - retry instead
         set +e
+
+        # get the password from the aws client
+        password=${bashDollar}(awscliv2 ecr get-login-password --region ${bashDollar}{AWS_REGION})
```

Comment:
Shall we add something like 

`if [[ $? -ne 0 ]]; then
    echo "Could not get password from AWS"
    continue
fi`

To prevent unnecessary docker login errors?

Reply:
done

Reply:
We still don't "continue" if the result value is not 0, also I am not wondering if the echo message won't be confusing as we are going to retry anyway.